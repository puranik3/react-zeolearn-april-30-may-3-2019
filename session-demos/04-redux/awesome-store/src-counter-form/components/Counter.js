import React from 'react';

function Counter( props ) {
    const { counter, increment, decrement } = props;
    
    return (
        <div>
            <p>{counter}</p>
            <button onClick={increment}>+</button>
            <button onClick={decrement}>-</button>
        </div>
    );
}

export default Counter;