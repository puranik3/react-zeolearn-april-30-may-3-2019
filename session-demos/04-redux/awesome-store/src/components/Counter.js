import React from 'react';

function Counter( props ) {
    const { increment, decrement, counter } = props;
    
    return (
        <div>
            <p>{counter}</p>
            <button onClick={increment}>+</button>
            <button onClick={decrement}>-</button>
        </div>
    );
}

export default Counter;