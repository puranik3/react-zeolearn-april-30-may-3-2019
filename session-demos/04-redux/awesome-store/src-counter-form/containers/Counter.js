import Counter from '../components/Counter';
import { connect } from 'react-redux';
import { incrementAC, decrementAC } from '../actions/creators';

const mapStateToProps = state => ( { counter: state.counter } );
const mapDispatchToProps = dispatch => ({
    increment: () => dispatch( incrementAC() ),
    decrement: () => dispatch( decrementAC() )
});

export default connect( mapStateToProps, mapDispatchToProps )( Counter );