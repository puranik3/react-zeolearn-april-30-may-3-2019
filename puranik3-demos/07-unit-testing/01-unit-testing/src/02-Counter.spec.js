import { shallow } from 'enzyme';
import React from 'react';
import Counter from './02-Counter';
import renderer from 'react-test-renderer';

// @todo Check counter starts with state 0
describe( 'Counter', function() {
    let wrappedCounter;

    beforeEach(function() {
        wrappedCounter = shallow( <Counter /> );
    })

    it( 'starts with a state.count 0', function() {
        expect( wrappedCounter.state().count ).toBe( 0 );
    });

    it( 'increases count by 1 when increment button is clicked', function() {
        wrappedCounter.find( '#increment' ).simulate( 'click' );
        expect( wrappedCounter.state().count ).toBe( 1 );
    });
    
    it( 'decreases count by 1 when decrement button is clicked', function() {
        wrappedCounter.find( '#decrement' ).simulate( 'click' );
        expect( wrappedCounter.find( '#counter' ).text() ).toBe( "-1" );
    });


});

// @todo Create and check against snapshot (renderer.create( <Counter /> ).toJSON())