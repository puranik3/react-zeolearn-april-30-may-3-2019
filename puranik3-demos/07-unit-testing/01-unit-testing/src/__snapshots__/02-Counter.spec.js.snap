// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should match snapshot 1`] = `
<div>
  Counter: 
  <span
    id="counter"
  >
    0
  </span>
  <button
    id="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <button
    id="increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
